packages/next-auth/src/client/_utils.ts

```diff
@@ -94,10 +94,18 @@ export function BroadcastChannel(name = "nextauth.message") {
     /** Notify other tabs/windows. */
     post(message: Record<string, unknown>) {
       if (typeof window === "undefined") return
```

Comment:
```suggestion
      if (typeof localStorage !== "object") return
```

Should this not be enough? :thinking:

Reply:
This is not enough unfortunately, some browsers throw when calling the `setItem` and `getItem` methods, even if they both exist. Sadly the only reliable way to test if the API is available (and working as expected) is to do something like this:

```
function isSupported(getStorage) {
  try {
    const key = "something-random";
    getStorage().setItem(key, key);
    getStorage().removeItem(key);

    return true;
  } catch (e) {
    return false;
  }
}
```

https://michalzalecki.com/why-using-localStorage-directly-is-a-bad-idea/

Reply:
:exploding_head: Thanks for the context!